### ISSUE TITLE
import not working the same with ts-node and tsc + node

### ISSUE THREAD

**Issue by bjg2** [TypeStrong/ts-node#786]
ts-node version v8.0.2
tsc version 3.3.3

Please take a look at [ts-node-fail](https://github.com/bjg2/ts-node-fail) repo, there is a small repro.

If I execute
```
ts-node test.ts
```
while there are no transpiled js files (so there's no config.js) config.ts import is messed up and I get output:
```
hi from test!
```

If I execute
```
tsc test.ts
node test.js
```
import works as expected and I get full output
```
hi from config!
hi from test!
```

Am I using something wrong? Seems like ts-node just has the bad execution.

**Comment by 3mard** [comment]
To understand why is this happening you should understand how[ nodejs resolves files](https://nodejs.org/api/modules.html#modules_file_modules) 

> File Modules#
If the exact filename is not found, then Node.js will attempt to load the required filename with the added extensions: .js, .json, and finally .node.

Now let's examine what happens when you use tsc:
`tsc test.ts --outDir dist`

`ls`

```
.
├── config.js
├── config.json
├── config.ts
├── package-lock.json
├── package.json
├── test.js
└── test.ts
```

Now according to the node docs when you try to `import ./config` it will first try to import config.js and if not found it will try to import config.json, it will find config.js and thus  'hi from config!' will be prinited

Now let's examine how ts-node work :
Basically [ts-node registesr an extension handler for [ts,tsx] files ](https://github.com/TypeStrong/ts-node/blob/master/src/index.ts#L404) 

so require.extensions becomes [ '.js', '.json', '.node', '.ts', '.tsx' ]   and according to the order of th array `import './config'` will import the json file not the typescript file

**Comment by bjg2** [comment]
Is it possible to change the order of the files in require.extension, so that '.ts' extension is just after '.js'? I guess the idea of ts-node is to have exactly the same behavior as tsc + node, this bug was subtle and it actually took me a lot of time to find it because I didn't expect that it works differently.

**Comment by 3mard** [comment]
I think it's a bad practice to use such kind of ambiguous setup , [and it was a bad design decision of nodejs to allow this ambiguity ](https://www.youtube.com/watch?v=M3BM9TB-8yA)

Best to my knowledge #649  is trying fix this

**Comment by bjg2** [comment]
Yeah, I will rename files and thanks for the video. :) Moved the conversation to the issue mentioned (https://github.com/TypeStrong/ts-node/issues/648).

**Comment by blakeembrey** [comment]
Closing with https://github.com/TypeStrong/ts-node/pull/837.
